server/portal/apps/datafiles/views.py

```diff
@@ -64,7 +67,14 @@ def get(self, request, operation=None, scheme=None, system=None, path='/'):
         try:
             client = request.user.agave_oauth.client
         except AttributeError:
-            client = None
+            # Make sure that we only let unauth'd users see public systems
+            if next(sys for sys in settings.PORTAL_DATAFILES_STORAGE_SYSTEMS
+                    if sys['system'] == system and sys['scheme'] == 'public'):
+                client = service_account()
+            else:
+                return JsonResponse(
+                    {'message': 'This data requires authentication to view.'},
+                    status=403)
```

Comment:
Not sure if anyone else ran into this issue, or if it is even related to this PR, but when previewing files I found that the `tapis_get_handler` is passed `request.GET.dict()` as kwargs. Further on in `/server/portal/libs/agave/operations.py` a string value within the object passed to kwargs is compared against an int `kwargs['length'] < 10000000` which causes a python error and the preview to fail for `.txt` files specifically.

Reply:
Thanks for catching this. I pushed a fix to coerce those values to integers.